src/XMakeBuildEngine/Definition/ProjectCollection.cs

```diff
@@ -817,11 +812,9 @@ internal int MaxNodeCount
 
         /// <summary>
         /// The cache of project root elements associated with this project collection.
-        /// Each is associated with a specific project collection for two reasons:
-        /// - To help protect one project collection from any XML edits through another one:
-        /// until a reload from disk - when it's ready to accept changes - it won't see the edits;
```

Comment:
I am a little bit concerned about losing this property, but the benefit seems worth it. Cases where something is sneakily updating imports during a build causing torn state seem like they'd be rare.